src/init.cpp

```diff
@@ -1439,22 +1439,26 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     for (bool fLoaded = false; !fLoaded && !ShutdownRequested();) {
         node.mempool = std::make_unique<CTxMemPool>(mempool_opts);
 
+        const bool fReset = fReindex;
+
         const ChainstateManager::Options chainman_opts{
             .chainparams = chainparams,
             .adjusted_time_callback = GetAdjustedTime,
+            .block_tree_db_opts = {
+                .db_path = gArgs.GetDataDirNet() / "blocks" / "index",
```

Comment:
1ed5723bc9c69bf2d7c2a902e44f3fc46b1165a7: It is highly illegal to use `gArgs` in init.cpp in new code, when `args` can be used.

Reply:
Fixed in ed7ae6a08d 😅